### ISSUE TITLE
Sanity check environments before pushing to default channel

### ISSUE THREAD

**Issue by shughes-uk** [coiled/software-environments#33]
Rather than just doing create_software_environment and calling it a day, we should first create the environment on a private account and verify it works with the cluster quickstart. This should act as a good sanity check and avoid us pushing environments that build but do not actually function.

**Comment by ian-r-rose** [comment]
Agreed. As an alternative (not necessarily mutually exclusive), we could split up the staging and prod builds and just verify the staging ones before merging the prod ones. That would test against a wider array of things.

**Comment by shughes-uk** [comment]
Do the integration tests use much of the default environments? They'd be a good sanity check on staging

**Comment by ian-r-rose** [comment]
Yes, the integration tests use the default environment for the same python version that the client is running (today python3.8)

**Comment by ian-r-rose** [comment]
Another thought about testing against `staging` first: at one point our default environments were shared between staging and prod.  Do you know if that is still the case @dantheman39 ?
